docs/design/design.md

```diff
@@ -0,0 +1,32 @@
+3. communicating the Installer's design goals and principles while colloborating with other teams
+
+# Design Documentation
+
+This is an attempt to document the OpenShift Installer's existing design principles and also articulate design goals to guide with new feature development and in some cases inform bug fixes. This is expected to be a living document which is continually updated when we gain clarity around some of these principles.
+
+## Design Goals
+
+### Install methods
+
+The OpenShift Installer currently supports 2 installation types, UPI (User Provisioned Infrastructure) and IPI (Installer Provisionined Infrastructure). IPI is an opinionated method of installing OpenShift. The IPI installation method tries to automate a large portion of the OpenShift Install with the least amount of intervention from the user. While it tries to take care of supporting the most common install scenarios, it cannot be expected to support every install variation that a user wants to use for their cloud.
+
+This is where the UPI offers the greater flexibility required for these highly specialized installs. In this install method, the user is expected to provide several Infrastructure components needed for the install on their own. For this reason, documentation surrounding the UPI flow are especially important.
+
+The OpenShift Installer has the responsibility to validate configuration provided to it. There is a validation step that is common to both the UPI and IPI install methods and additional validation is performed specially for the IPI install method.
+
+### Cluster Configuration
+
+It is a design goal for the Installer to obtain configuration from the user in the most user-friendly way. One of the ways to accomplish that goal would be to request a minimal amount of configuration to create an OpenShift cluster on the platform and with the capabilities requested by the user. The Installer and OpenShift as a whole, have to strive to define sane defaults for configuration whenever possible and allow the user to override these defaults.
+
+Another guiding principle to consider regarding configuration during install, would be that any information that can be learned from the install environment should not be requested as an input from the user. And, if another OpenShift entity is capable of learning this information, the Installer would defer this responsibility to this entity. This prevents the Installer from becoming a monolith and allows this component to keep its treatment of this environmental config consistent during the time of cluster bringup and day-2.
+
+When adding any new configuration for enabling a new functionality within the OpenShift cluster, in addition to designing for a new Installation, consider how a Cluster upgraded to the version supporting this feature would behave. The design goal here would be for a cluster upgraded to a version supporting this feature behave identically to a fresh Installed cluster of that version.
```

Comment:
Can we figure out a way to flip these sentences ?

Reply:
Attempted.